store.go

```diff
@@ -2836,6 +2836,13 @@ func (s *store) mount(id string, options drivers.MountOpts) (string, error) {
 	if rlstore.Exists(id) {
 		return rlstore.Mount(id, options)
 	}
+	// check if the layer is in a read-only store, and return a better error message
+	for _, store := range lstores {
+		if store.Exists(id) {
```

Comment:
This requires `store.startReading`.

(A possible micro-optimization would be to end the `stopWriting` scope before this new code, to improve chance of concurrent accesses.)

Reply:
thanks.  Opened a follow up PR: https://github.com/containers/storage/pull/1877